content/blog/2024-02-26-mainnet-hardfork.mdx

```diff
@@ -0,0 +1,60 @@
+image: /images/blog/mainnet_hardfork.png
+tags: [ironfish, iron fish, hardfork, hard-fork, hard fork, activation, mainnet]
+---
+
+# Mainnet Activation for Hardfork-1
+
+Iron Fish is preparing for its first hardfork. The first hardfork will include:
+- [FIP-2](https://fips.ironfish.network/fips/fip-2-enforce-sequential-block-time) : Enforce sequential block time (a block’s timestamp has to be greater than the timestamp of its parent) 
+- [FIP-3](https://fips.ironfish.network/fips/fip-3-memory-hard-mining-algorithm) : Switch hashing algorithm from blake3 to FishHash
+- [FIP-8](https://fips.ironfish.network/fips/fip-8-difficulty-adjustment) : Difficulty adjustment (to accommodate the algorithm change)
+- [FIP-9](https://fips.ironfish.network/fips/fip-9-swap-randomness-and-graffiti) : Swap randomness and graffiti field positions in the Block Header
+
+The most notable change is the hashing algorithm switch from blake3 to FishHash. To read more about FishHash, the motivation behind finding a new hashing algorithm for Iron Fish, as well as exploration of other algorithm choices, please see [FIP-3](https://fips.ironfish.network/fips/fip-3-memory-hard-mining-algorithm) (or community created video walkthroughs like this [one](https://youtu.be/1R7VffTzM9g?si=U0PhustQsp8TG9ak)). Please note that [FIP-9](https://fips.ironfish.network/fips/fip-9-swap-randomness-and-graffiti) that is included in this hardfork also changes the Block Header format for Iron Fish. 
+
+The timeline for the mainnet hardfork is as follows: 
+
+| status | date | milestone |
+| --- | --- | --- |
+| deployed ✅  | Feb 20th | Release of the Iron Fish node version that includes Testnet hardfork activation (see [release notes and binaries](https://github.com/iron-fish/ironfish/releases/tag/v1.20.0)) |
+| deployed ✅  | Feb 28th | [Testnet](https://testnet.explorer.ironfish.network/) hardfork activation at block sequence 419,193 |
+| deployed ✅  | March 6th | Release of the Iron Fish node version that includes Mainnet hardfork activation (see [release notes and binaries](https://github.com/iron-fish/ironfish/releases/tag/v2.0.0)) |
+| upcoming 👉 | April 2nd | [Mainnet](https://explorer.ironfish.network/) hardfork activation at block sequence 503,338, estimated April 2 2024 12:00 GMT |
+
+If you’re an Iron Fish user, developer, wallet provider, miner, exchange, or anyone in between, please make sure to **upgrade your Iron Fish Node to [version 2.0.0](https://github.com/iron-fish/ironfish/releases/tag/v2.0.0) (or greater if available) by no later than April 2nd to ensure a smooth transition for all Iron Fish users.** If you’ve used `npm` previously to install Iron Fish node you can upgrade by simply running the same command again: `npm install -g ironfish`.
+
+If you’re using the Node App, please make sure to upgrade it as well to **[version 3.0](https://ironfish.network/use/node-app) (or greater if available).** 
+
+### What you need to know if you’re a miner or a mining pool
```

Comment:
I think we should have separate sections for miners and mining pools. Miners are a big cohort and mostly don't run a node whereas mining pools run a node. So very different instructions. For miners I like something like

```
### What you need to know if you’re a miner
At block 503,338 (~April 2nd) miners will need to manually switch their devices to start mining FishHash instead of blake3. Your mining pool should already be compatible with the change. All [major mining software](https://ironfish.network/use/get-started/mining#mining-software) developers are planning on releasing new versions that include a FishHash miner. Be on the lookout for a new release of your mining software that includes FishHash which should be available at least a week before the hardfork. 
```

Reply:
For pools maybe something like

```
### What you need to know if you’re a mining pool
You need to **upgrade** your Iron Fish node **as soon as possible** to [version 2.0.0](https://github.com/iron-fish/ironfish/releases/tag/v2.0.0). Some notable changes for pools:
 - [FIP-2](https://fips.ironfish.network/fips/fip-2-enforce-sequential-block-time) : If your pool creates blocks with negative timestamps those will no longer be valid after the hardfork
 - [FIP-3](https://fips.ironfish.network/fips/fip-3-memory-hard-mining-algorithm) : The obvious switch of the mining algorithm from blake3 to FishHash
- [FIP-9](https://fips.ironfish.network/fips/fip-9-swap-randomness-and-graffiti) :  randomness and graffiti field positions are switched in the Block Header

All these changes are included in the Node v2.0.0 release but if you have customized your node make sure that those changes are still compatible
```